src/test_snmp.cpp

```diff
@@ -282,6 +283,46 @@ TEST_F(SNMPTest, IPCountTable)
   delete tbl;
 }
 
+TEST_F(SNMPTest, SuccessFailCountTable)
+{
+  // Create table
+  SNMP::SuccessFailCountTable* tbl = SNMP::SuccessFailCountTable::create("success_fail_count", test_oid);
+  
+  // Shell out to snmpwalk to find all entries in that table
+  FILE* fd;
+  char buf[1024];
+  
+  tbl->increment_attempts();
+  tbl->increment_successes();
+  
+  // Should be 1 attempt, 1 success, 0 failures.
+  fd = popen("snmpwalk -v2c -On -c clearwater 127.0.0.1:16161 .1.2.2.1.2.2", "r");
+  fgets(buf, sizeof(buf), fd);
+  ASSERT_STREQ(".1.2.2.1.2.2 = Gauge32: 1\n", buf);
+  fd = popen("snmpwalk -v2c -On -c clearwater 127.0.0.1:16161 .1.2.2.1.3.2", "r");
+  fgets(buf, sizeof(buf), fd);
+  ASSERT_STREQ(".1.2.2.1.3.2 = Gauge32: 1\n", buf);
+  fd = popen("snmpwalk -v2c -On -c clearwater 127.0.0.1:16161 .1.2.2.1.4.2", "r");
+  fgets(buf, sizeof(buf), fd);
+  ASSERT_STREQ(".1.2.2.1.4.2 = Gauge32: 0\n", buf);
+
+  tbl->increment_attempts();
+  tbl->increment_failures();
+  
+  // Should be 2 attempts, 1 success, 1 failure.
+  fd = popen("snmpwalk -v2c -On -c clearwater 127.0.0.1:16161 .1.2.2.1.2.2", "r");
+  fgets(buf, sizeof(buf), fd);
+  ASSERT_STREQ(".1.2.2.1.2.2 = Gauge32: 2\n", buf);
```

Comment:
What happens to this test if your first `increment_attempts` call was at 10:04:59.999 and your second was at 10:05:00.002?